app/controllers/auth/registrations_controller.rb

```diff
@@ -76,4 +95,18 @@ def password_update?
     !user[:current_password].blank? || !user[:password].blank? ||
       !user[:password_confirmation].blank?
   end
+
+  # Returns whether the given user can be disabled or not. The following rules
+  # apply:
+  #   1. A user can disable himself unless it's the last admin on the system.
+  #   2. The admin user is the only one that can disable other users.
+  def can_disable?(user)
+    if current_user == user
+      # An admin cannot disable himself if he's the only admin in the system.
+      current_user.admin? && User.admins.count == 1
+    else
+      # Only admin users can disable other users.
+      !current_user.admin?
```

Comment:
Note that a regular user should be able to disable himself, therefore non-admin users should be able to perform this action. I'd say to just remove `:disable` from the `check_admin` hook (which by the way is not preventing anything, since the test in line 145 of spec/controllers/auth/registrations_controller_spec.rb passes ;) ).